src/NServiceBus.Callbacks/RequestResponseExtensions.cs

```diff
@@ -0,0 +1,102 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Configuration.AdvanceExtensibility;
+
+    /// <summary>
+    /// Synchronous request/response extension methods.
+    /// </summary>
+    public static class RequestResponseExtensions
+    {
+        /// <summary>
+        /// Sends a <paramref name="requestMessage"/> to the configured destination and returns back a <see cref="SendContext{TResponse}"/> that allows the use user to wait on.
+        /// </summary>
+        /// <typeparam name="TResponse">The response type.</typeparam>
+        /// <param name="bus">Object beeing extended.</param>
+        /// <param name="requestMessage">The request message.</param>
+        /// <param name="options">The options for the send.</param>
+        /// <returns>A synchronous request/response context.</returns>
+        public static SendContext<TResponse> SynchronousRequestResponse<TResponse>(this IBus bus, object requestMessage, SynchronousOptions options)
+        {
+            if (requestMessage == null)
+            {
+                throw new ArgumentNullException("requestMessage");
+            }
+
+            if (options == null)
+            {
+                throw new ArgumentNullException("options");
+            }
+
+            if (bus == null)
+            {
+                throw new ArgumentNullException("bus");
+            }
+
+            var sendOptions = new SendOptions(options.Destination, options.CorrelationId);
+                
+            foreach (var header in options.Headers)
+            {
+                sendOptions.AddHeader(header.Key, header.Value);
+            }
+
+            sendOptions
+                .AddHeader("NServiceBus.HasCallback", Boolean.TrueString)
+                .SetCustomMessageId(options.MessageId);
+
+            var tcs = new TaskCompletionSource<TResponse>();
+
+            sendOptions.GetContext().Add("NServiceBus.RequestResponse.TCS", tcs);
```

Comment:
Heads up: After my `publish-options` pull this will need to be strongly typed:

```
sendOptions.GetContext().Set(new RequestResponseContext{ TaskCompletionSource = tcs })
```

cc @SzymonPobiega